pkg/controller/sync/sync_controller.go

```diff
@@ -103,8 +98,10 @@ func (r *ReconcileSync) Reconcile(request reconcile.Request) (reconcile.Result,
 	err := r.Get(context.TODO(), request.NamespacedName, instance)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			// Object not found, return.  Created objects are automatically garbage collected.
-			// For additional cleanup logic use finalizers.
+			// This is a deletion; remove the data
+			if _, err := r.opa.RemoveData(context.Background(), instance); err != nil {
```

Comment:
We need to manually set namespace and name on `instance` in this case so that the system can figure out which exact item to delete. IIRC the unstructured resource has methods on it to make this easy.

We should also make sure that a failed Get() leaves `instance` unaltered. If this can't be verified we can just recreate instance